### PR TITLE
Collate and list resources about the dockershim removal

### PR DESCRIPTION
**Pull request by rolfedh**
For https://github.com/kubernetes/website/issues/30919

### CODE REVIEW

File: content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+reviewers:
+- tbd
+title: Topics on dockershim and CRI-compatible runtimes
+content_type: concept
+weight: 20
+---
+<!-- overview -->
+The following topics cover the dockershim deprecation and CRI-compatible runtimes."
```
Comment (jihoon-seo) [inline content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md:9]:
```suggestion
The following topics cover the dockershim deprecation and CRI-compatible runtimes.
```

File: content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+<!-- body -->
+
+## Primary sources
+
+[https://github.com/kubernetes/enhancements/issues/2221](Removing dockershim from kubelet #2221)
+
+[https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2221-remove-dockershim/README.md](KEP-2221: Removing dockershim from kubelet)
```
Comment (jihoon-seo) [inline content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md:16]:
```suggestion
[Removing dockershim from kubelet (`kubernetes/enhancements#2221`)](https://github.com/kubernetes/enhancements/issues/2221)

[KEP-2221: Removing dockershim from kubelet](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2221-remove-dockershim/README.md)
```

File: content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+[Kubernetes Blog: Dockershim Deprecation FAQ](https://kubernetes.io/blog/2020/12/02/dockershim-faq/)
+
+[Kubernetes Documentation: Migrating from dockershim](https://kubernetes.io/docs/tasks/administer-cluster/migrating-from-dockershim/)
+
+[Kubernetes Documentation: Container runtimes](https://kubernetes.io/docs/setup/production-environment/container-runtimes/)
```
Comment (jihoon-seo) [inline content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md:22]:
```suggestion
[Kubernetes Blog: Dockershim Deprecation FAQ](/blog/2020/12/02/dockershim-faq/)

[Kubernetes Documentation: Migrating from dockershim](/docs/tasks/administer-cluster/migrating-from-dockershim/)

[Kubernetes Documentation: Container runtimes](/docs/setup/production-environment/container-runtimes/)
```

File: content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+
+## Secondary sources
+
+* [Docker blog: What developers need to know about Docker, Docker Engine, and Kubernetes v1.20](https://www.docker.com/blog/what-developers-need-to-know-about-docker-docker-engine-and-kubernetes-v1-20/)
```
Comment (jihoon-seo) [inline content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md:27]: For consistency:
- Remove the bullet here, or
- Add bullets for the `Primary sources` entries

File: content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+---
+reviewers:
+- tbd
```
Comment (jihoon-seo) [inline content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md:3]:
```suggestion
```

**Review by jihoon-seo — COMMENTED** [review]
Preview: https://deploy-preview-31528--kubernetes-io-main-staging.netlify.app/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes/

@rolfedh Thank you for your contribution!
I added some comments.
PTAL! 😊

File: content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md (hunk 5 above)
Comment (rolfedh) [reply content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md:3]: Thanks, @jihoon-seo! Will update.

File: content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+title: Topics on dockershim and CRI-compatible runtimes
+content_type: concept
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md:5]: We usually don't put concept pages in the tasks section.

Does this belong inside https://k8s.io/docs/reference/ perhaps?

**Review by sftim — COMMENTED** [review]
Some early feedback.

- I'd make this a reference page
- Early in the page, we should link to `/dockershim` (specifically the short URL redirect - that way, if we change where the redirect points, we pick up that change automatically)
- I'd link to 2 extra pages:
  - https://www.mirantis.com/blog/the-future-of-dockershim-is-cri-dockerd/
  - https://github.com/Mirantis/cri-dockerd

File: content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md (hunk 6 above)
Comment (rolfedh) [reply content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md:5]: Thanks, @sftim. I don't really understand the k8s docs' paradigm for organizing content by content type. Maybe we could huddle on this briefly in a day or so? 

I agree that this "list-of-links" topics seems more like a reference topic, so I've updated this line to:  
```suggestion
content_type: reference
```
Where should it go within the `docs/reference` folder?

**Comment by shannonxtreme** [comment]
It would definitely be nice to organize our docs by what our readers are
looking to accomplish. For example "Secure your clusters and workloads"
with security related conceptual content and tasks. Could be an interesting
discussion at a SIG meeting 😊 it would need significant resources from our
end I think but still worth talking about!

File: content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md (hunk 2 above)
Comment (rolfedh) [reply content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md]: Updated, thanks.

File: content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md (hunk 6 above)
Comment (rolfedh) [reply content/en/docs/tasks/administer-cluster/migrating-from-dockershim/topics-on-dockershim-and-cri-compatible-runtimes.md]: Thank you, @sftim. PTAL and let me know what additional changes might be needed.

**Comment by rolfedh** [comment]
> It would definitely be nice to organize our docs by what our readers are looking to accomplish. For example "Secure your clusters and workloads" with security related conceptual content and tasks. Could be an interesting discussion at a SIG meeting  it would need significant resources from our end I think but still worth talking about!

Thanks, @shannonxtreme. I think you're commenting on the overall content and file structure of the k8s docs, rather than about this specific PR. If so, I'd recommend reviewing the [kubernetes/website issues](https://github.com/kubernetes/website/issues) and then, if it isn't already covered, creating a **Feature request** for your suggestion. Then participate in the SIG meetings and bring it up!

**Comment by shannonxtreme** [comment]
Nope yeah just a miscommunication lol I thought you were talking about the
overall docs site. Thanks!

**Comment by rolfedh** [comment]
Thanks again, @jihoon-seo and @sftim. I believe the current PR reflects all your suggestions. The only open question is where or how to place the file correctly within the `/reference` directory.  Any thoughts?

**Comment by sftim** [comment]
> where or how to place the file correctly within the `/reference` directory.

https://kubernetes.io/docs/reference/node/dockershim-removal/ ? Just my suggestion.

https://kubernetes.io/docs/reference/node/ can be the barest placeholder, but does need at least a title (eg “Node Reference Information”)

File: content/en/docs/reference/topics-on-dockershim-and-cri-compatible-runtimes.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
+---
+title: Topics on dockershim and CRI-compatible runtimes
+content_type: reference
+weight: 20
+---
+<!-- overview -->
+Articles about the Kubernetes' deprecation of dockershim, and using CRI-compatible runtimes.
```
Comment (sftim) [inline content/en/docs/reference/topics-on-dockershim-and-cri-compatible-runtimes.md:7]:
```suggestion
This is a list of articles about:
- the Kubernetes' deprecation and removal of _dockershim_
- using CRI-compatible container runtimes
```

The key part for me is “and removal”. I want readers who aren't clear on the meaning of “deprecation” to infer that it's something that happens before removal.

Comment (rolfedh) [reply content/en/docs/reference/topics-on-dockershim-and-cri-compatible-runtimes.md:7]: Good point! Updated. Thanks.

**Comment by rolfedh** [comment]
> https://kubernetes.io/docs/reference/node/ can be the barest placeholder, but does need at least a title (eg “Node Reference Information”)

@sftim - like this?

File: content/en/docs/reference/dockershim-removal/_index.md [hunk 8]
```diff
@@ -0,0 +1,4 @@
+---
+title: Dockershim Removal and CRI-compatible Runtimes
```
Comment (sftim) [inline content/en/docs/reference/dockershim-removal/_index.md:2]: How I intended it

```suggestion
title: Node Reference Information
```
_and_
change the path:
```bash
mv content/en/docs/reference/dockershim-removal content/en/docs/reference/node/
```


If we have future node-related references (I expect that we will), they can also live in here.

File: content/en/docs/reference/dockershim-removal/topics-on-dockershim-and-cri-compatible-runtimes.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
+---
+title: Topics on dockershim removal and CRI-compatible runtimes
```
Comment (sftim) [inline content/en/docs/reference/dockershim-removal/topics-on-dockershim-and-cri-compatible-runtimes.md:2]: (nit) Should be title case. Eg
```suggestion
title: External Articles on dockershim Removal and on Using CRI-compatible Runtimes
```
?

**Comment by rolfedh** [comment]
Thanks, @sftim. It think this might be ready for approval and merge...

File: content/en/docs/reference/node/topics-on-dockershim-and-cri-compatible-runtimes.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+* [Kubernetes Documentation: "Migrating from dockershim"](/docs/tasks/administer-cluster/migrating-from-dockershim/)
+
+* [Kubernetes Documentation: "Container runtimes"](/docs/setup/production-environment/container-runtimes/)
+
+* [Kubernetes enhancement issue: "Removing dockershim from kubelet" (`kubernetes/enhancements#2221`)](https://github.com/kubernetes/enhancements/issues/2221)
+
+* [Kubernetes enhancement proposal: "KEP-2221: Removing dockershim from kubelet"](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2221-remove-dockershim/README.md)
+
+* [Kubernetes Blog: "Dockershim Deprecation FAQ", 2020/12/02](/blog/2020/12/02/dockershim-faq/)
```
Comment (sftim) [inline content/en/docs/reference/node/topics-on-dockershim-and-cri-compatible-runtimes.md:22]: nit: I'd put this one right at the top.

File: content/en/docs/reference/node/topics-on-dockershim-and-cri-compatible-runtimes.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
+---
+title: External Articles on dockershim Removal and on Using CRI-compatible Runtimes
+content_type: reference
+weight: 20
+---
+<!-- overview -->
+This is a list of articles about:
+	- the Kubernetes' deprecation and removal of _dockershim_
+	- using CRI-compatible container runtimes
```
Comment (sftim) [inline content/en/docs/reference/node/topics-on-dockershim-and-cri-compatible-runtimes.md:9]: Oops.
```suggestion
This is a list of articles about:

- the Kubernetes' deprecation and removal of _dockershim_
- using CRI-compatible container runtimes
```

**Review by sftim — COMMENTED** [review]
Two more things, one an optional fix

**Comment by sftim** [comment]
Relevant to https://github.com/kubernetes/website/issues/30839

**Comment by jihoon-seo** [comment]
/retitle Dockershim: collate a list of 3rd party resources on the topic

**Comment by rolfedh** [comment]
> /retitle Dockershim: collate a list of 3rd party resources on the topic

Hi @jihoon-seo. The document contains both primary (Kubernetes.io) and third-party sources.

**Comment by rolfedh** [comment]
> /retitle Dockershim: collate a list of 3rd party resources on the topic

Hi @jihoon-seo. The document contains both primary (Kubernetes.io) and third-party sources.

**Comment by sftim** [comment]
If this is no longer WIP @rolfedh, feel free to update the PR title accordingly.

**Comment by rolfedh** [comment]
@sftim Next steps to get this PR approved and merged?

**Comment by sftim** [comment]
/approve

Thanks @rolfedh and @nate-double-u

**Pull request merged**